### PR TITLE
3d fft

### PR DESCRIPTION
**Pull request by geoffwoollard**
Fixes https://github.com/compSPI/compSPI/issues/33.

Closely follows 2D functions and tests, with small changes.

Shape of 2d fft is (batch,1,size,size). Not sure about the 1 in dim=1.

Shape of 2d fft is (batch, size, size, size). Not (batch, 1, size, size, size). But this could be done. I'm just not sure what the extra axis is for... Channel / RGB?

### CODE REVIEW

**Comment by geoffwoollard** [comment]
@harshit-gupta-cor I added you since you wrote the 2d fft wrappers

File: .flake8 [hunk 1]
```diff
@@ -2,5 +2,5 @@
 docstring-convention = numpy
 import_order_style = smarkets
 max-line-length = 88
-extend-ignore = E203,I202,I100
+extend-ignore = E203,I202,I100,D410,D411
```
Comment (geoffwoollard) [inline .flake8:5]: I really can't figure out why I need these flags. I'm not sure how to resolve them. As far as I can see the docstring structure is exactly the same between the old 2D and new 3D functions. But for some reason there are errors

```
transforms.py:44:1: D410 Missing blank line after section
transforms.py:44:1: D411 Missing blank line before section
transforms.py:63:1: D410 Missing blank line after section
transforms.py:63:1: D411 Missing blank line before section
```

File: tests/test_transforms.py [hunk 2]
```diff
@@ -2,11 +2,14 @@
 
 import torch
 
-from compSPI.transforms import fourier_to_primal_2D, primal_to_fourier_2D
+from compSPI.transforms import fourier_to_primal_2D as ifft_2d
```
Comment (geoffwoollard) [inline tests/test_transforms.py:5]: I called them something else just to avoid importing them all in one line, which was causing issues with a black and isort clash. 

There was issues with a long line of imports, and I couldn't just put them on separate lines, because they would get put together by the precommit hook. So I used `as` to force them to be separate import lines.

File: tests/test_transforms.py [hunk 3]
```diff
@@ -31,21 +31,69 @@ def test_primal_to_fourier_2D():
...
     constant and delta, respectively.
     """
     for size in [50]:
+
         im_fourier = torch.zeros(2, 1, size, size)
         im_fourier[0, 0, size // 2, size // 2] = 1
         im_fourier[1, 0, :, :] = 1
 
-        im = fourier_to_primal_2D(im_fourier)
+        im = compSPI.transforms.fourier_to_primal_2D(im_fourier)
```
Comment (geoffwoollard) [inline tests/test_transforms.py:45]: I didn't import the four functions tested, because there was some clash with isort and black that I couldn't resolve. They were formatting the long line of the import in different and incompatible ways.

**Comment by geoffwoollard** [comment]
Fixes #33

**Comment by geoffwoollard** [comment]
@ninamiolane @fredericpoitevin OK to merge?

**Comment by geoffwoollard** [comment]
Ok I'm just going to merge since all tests are passing.
